coderd/coderdtest/coderdtest.go

```diff
@@ -143,6 +143,8 @@ func newWithCloser(t *testing.T, options *Options) (*codersdk.Client, io.Closer)
 	}
 
 	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer t.Cleanup(cancelFunc) // Defer to ensure cancelFunc is executed first.
```

Comment:
And here

Reply:
Since this is a helper function, `defer cancelFunc()` would run before the calling test gets a chance to do anything with it. And a `t.Cleanup` without `defer` would cancel the context too late, it would not trigger teardown. So I realize this is a bit of a sneaky way to do it, but this way `t.Cleanup(cancelFunc)` will be left on top of the cleanup stack when this function exits, triggering the teardown we want (and same order as before the refactor).